hibernate-core/src/main/java/org/hibernate/persister/collection/BasicCollectionPersister.java

```diff
@@ -699,7 +705,7 @@ private void applyDeleteRowRestrictions(
 						jdbcValueBindings,
 						null,
 						(valueIndex, bindings, noop, jdbcValue, jdbcValueMapping) -> {
-							if ( jdbcValueMapping.isNullable() || jdbcValueMapping.isFormula() ) {
+							if ( !elementColumnIsInPrimaryKey[valueIndex] || jdbcValueMapping.isFormula() ) {
```

Comment:
The change is unnecessary, because `!elementColumnIsInPrimaryKey[valueIndex] == !jdbcValueMapping.isNullable()`. Please revert it.